File: src/main/java/com/capgemini/krk/services/course/CourseService.java
package com.capgemini.krk.services.course;

import com.capgemini.krk.TO.course.CourseTO;
import com.capgemini.krk.TO.course.CoursesTO;

public interface CourseService {

    CoursesTO findAll();

    CourseTO findById(int id);

    CoursesTO findByCoursesModuleId(int id);

    void delete(int id);

    void updateCourse(CourseTO course);

    CourseTO insertCourse(int coursesModuleId, CourseTO course);

}

File: src/main/client/src/app/course-mgmt/course.service.ts
import {Injectable} from '@angular/core';
import {Http, Response} from '@angular/http';
import 'rxjs/add/operator/map';
import {Course} from '../model/course.model';

@Injectable()
export class CourseService {

    constructor(private http: Http) {
    }

    /*findAll() {
      return this.http.get('http://localhost:8081/gui/studies-plans')
        .map((response: Response) => {
          console.log(response.json().studiesPlans);
          return response.json().studiesPlans;
        });
    } */

    findByCoursesModuleId(coursesModuleId: number) {
      return this.http.get('http://localhost:8081/gui/courses-overview/' + coursesModuleId)
        .map((response: Response) => {
          console.log(response.json().courses);
          return response.json().courses;
        });
    }

    findById(courseId: number) {
      return this.http.get('http://localhost:8081/gui/courses/' + courseId)
        .map((response: Response) => response.json());
    }

    updateCourse(course: Course) {
      return this.http.put('http://localhost:8081/gui/courses-up/', course);
    }

    insertCourse(coursesModuleId: number, course: Course) {
      return this.http.put('http://localhost:8081/gui/courses-in/' + coursesModuleId, course);
    }

    deleteCourse(id: number) {
      return this.http.delete('http://localhost:8081/gui/courses/' + id);
    }

}
